client/components/domains/map-domain-step/index.jsx

```diff
@@ -20,16 +20,17 @@ var MapDomainStep = React.createClass( {
 
 	propTypes: {
 		products: React.PropTypes.object.isRequired,
+		initialQuery: React.PropTypes.string,
 		analyticsSection: React.PropTypes.string.isRequired
 	},
 
 	getInitialState: function() {
-		return { searchQuery: '' };
+		return { searchQuery: this.props.initialQuery };
 	},
 
 	componentWillMount: function() {
 		if ( this.props.initialState ) {
-			this.setState( this.props.initialState );
+			this.setState( Object.assign( {}, this.props.initialState, this.getInitialState() ) );
```

Comment:
That's how the previously saved flow in signup is handled - I scratched my head around this too, but it's too big to fix here :/